druntime/posix.mak

```diff
@@ -350,8 +350,8 @@ $(IMPDIR)/%.h : src/%.h
 
 ######################## Build DMD if non-existent ##############################
 
-$(DMD):
-	$(MAKE) -C $(DMD_DIR)/src -f posix.mak BUILD=$(BUILD) OS=$(OS) MODEL=$(MODEL)
+$(DMD_DIR)/../generated/$(OS)/$(BUILD)/$(MODEL)/dmd:
+	$(MAKE) -C $(DMD_DIR)/.. dmd BUILD=$(BUILD) OS=$(OS) MODEL=$(MODEL) DMD=""
```

Comment:
In case `DMD` is set in the druntime `make` cmdline (semantics: use that DMD to compile druntime), it's automatically added as env variable, and affects the host-compiler lookup. If we get here to this recipe, `DMD` is the default one, *and* non-existent, so we obviously can't use that one as host compiler. So clear `DMD` explicitly here.